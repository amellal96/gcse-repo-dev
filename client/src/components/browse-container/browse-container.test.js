import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'
import { mount } from 'enzyme';

import BrowseContainer from './browse-container.component';

const mockStore = configureStore([thunk]);

describe('BrowseContainer component', () => {
    let store;
    let wrapper;

    beforeEach(() => {
        store = mockStore({
            question: jest.fn(),
            user: []
        });

        store.dispatch = jest.fn();

        const mockProps = {
            store
        }
        wrapper = mount(<BrowseContainer {...mockProps}/>)
    });

    // it('should match the snapshot', () => {
    //     expect(wrapper.html()).toMatchSnapshot(); 
    // });

    it('should render option fields', () => {
        // expect(wrapper.find('.option-container')).toHaveLength(4);
    });
});