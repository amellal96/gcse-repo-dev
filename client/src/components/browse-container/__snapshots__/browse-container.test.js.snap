// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseContainer component should match the snapshot 1`] = `"<div class=\\"browse-container\\"><div class=\\"filter-container\\"><div class=\\"container\\"><h3>Filter</h3><form class=\\"option\\"><h5>Exam Boards</h5><div class=\\"option-container\\"><div class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"AQA\\" value=\\"AQA\\"> AQA</div><div class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Edexcel\\" value=\\"Edexcel\\"> Edexcel</div><div class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"OCR\\" value=\\"OCR\\"> OCR</div><div class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"WJEC\\" value=\\"WJEC\\"> WJEC</div></div><h5>Topics</h5><div class=\\"option-container\\"><div class=\\"question-option-container\\"> <input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Numbers\\" value=\\"Number\\"> Number </div><div class=\\"question-option-container\\"> <input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Fractions\\" value=\\"Algebra\\"> Algebra </div><div class=\\"question-option-container\\"> <input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Ratio, Proportion and Rates of change\\" value=\\"Ratio, Proportion and Rates of change\\"> Ratio, Proportion and Rates of change </div><div class=\\"question-option-container\\"> <input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Geometry and Measures\\" value=\\"Geometry and Measures\\"> Geometry and Measures </div><div class=\\"question-option-container\\"> <input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Algebra\\" value=\\"Statistics &amp; Probability\\"> Statistics &amp; Probability </div></div><h5>Difficulty</h5><div class=\\"option-container\\"><label class=\\"col-form-label col-form-label-sm\\" for=\\"inputSmall\\">From</label><input name=\\"min\\" class=\\"form-control form-control-sm filter\\" type=\\"text\\" id=\\"inputSmall\\"><label name=\\"max\\" class=\\"col-form-label col-form-label-sm\\" for=\\"inputSmall\\">To</label><input class=\\"form-control form-control-sm filter\\" type=\\"text\\" id=\\"inputSmall\\"></div><h5>Marks</h5><div class=\\"option-container\\"><label class=\\"col-form-label col-form-label-sm\\" for=\\"inputSmall\\">From</label><input name=\\"min\\" class=\\"form-control form-control-sm filter\\" type=\\"text\\" id=\\"inputSmall\\"><label name=\\"max\\" class=\\"col-form-label col-form-label-sm\\" for=\\"inputSmall\\">To</label><input class=\\"form-control form-control-sm filter\\" type=\\"text\\" id=\\"inputSmall\\"></div><button type=\\"button\\" class=\\"btn btn-primary\\">Clear</button></form></div></div><table class=\\"table table-hover browse\\"><thead><tr><th>Question</th><th>Answer</th><th>Marks</th><th>Topics</th><th>Exam Boards</th><th>Grade</th><th>Rating</th><th>Save</th></tr></thead><tbody><tr class=\\"table-active\\"><td></td><td></td><td></td><td></td><td></td><td></td><td><div class=\\"vote-container\\"><span data-value=\\"1\\" class=\\"btn btn-outline-success\\">Upvote</span><span data-value=\\"-1\\" class=\\"btn btn-outline-danger\\">Downvote</span></div></td><td><div class=\\"unsave\\"><button type=\\"button\\" class=\\"btn btn-primary\\">Unsave</button></div></td></tr><tr class=\\"table-active\\"><td></td><td></td><td></td><td></td><td></td><td></td><td><div class=\\"vote-container\\"><span data-value=\\"1\\" class=\\"btn btn-outline-success\\">Upvote</span><span data-value=\\"-1\\" class=\\"btn btn-outline-danger\\">Downvote</span></div></td><td><div class=\\"save\\"><button type=\\"button\\" class=\\"btn btn-primary\\">Save</button></div></td></tr></tbody></table></div>"`;
